src/quo/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,178 @@
+(ns quo.components.wallet.transaction-progress.view
+  (:require [quo.components.buttons.button.view :as button]
+            [quo.components.icon :as icons]
+            [quo.components.markdown.text :as text]
+            [quo.components.tags.context-tag.view :as context-tag]
+            [quo.components.wallet.confirmation-progress.view :as confirmation-progress]
+            [quo.components.wallet.transaction-progress.style :as style]
+            [quo.foundations.colors :as colors]
+            [quo.theme :as quo.theme]
+            [react-native.core :as rn]
+            [utils.i18n :as i18n]))
+
+(defn- icon-internal
+  ([icon]
+   (icon-internal icon nil 20))
+  ([icon color]
+   (icon-internal icon color 20))
+  ([icon color size]
+   [rn/view {:style style/icon}
+    [icons/icon
+     icon
+     {:color    color
+      :size     size
+      :no-color (when (not color) true)}]]))
+
+(defn- text-internal
+  [{:keys [weight size accessibility-label color]
+    :or   {weight :semi-bold
+           size   :paragraph-1}}
+   title]
+  [text/text
+   {:accessibility-label accessibility-label
+    :ellipsize-mode      :tail
+    :number-of-lines     1
+    :weight              weight
+    :size                size
+    :style               {:color color}}
+   title])
+
+(defn- network-type-text
+  [state]
+  (cond
+    (or (= state :sending) (= state :pending))      (i18n/label :t/pending-on)
+    (or (= state :confirmed) (= state :finalising)) (i18n/label :t/confirmed-on)
+    (= state :finalized)                            (i18n/label :t/finalized-on)
+    (= state :error)                                (i18n/label :t/failed-on)))
+
+(defn- calculate-counter
+  [counter]
+  (if (< @counter 4) @counter "4"))
+
+(defn- text-steps
```

Comment:
I would consider putting "4" and "1" into constant vars so that it's easier to understand/maintain what's happening here, these can then be accessed using functions.
imo I would do something along the lines of:
```
(def ^:private max-mainnet-verifications 4)

(def ^:private max-sidenet-verifications 1)


(defn- calculate-counter
  [counter]
  (if (< counter 4) counter max-mainnet-verifications))


(defn mainnet-label [counter]
  (str counter "/" max-mainnet-verifications))

(defn subnet-label [counter]
  (str counter "/" max-sidenet-verifications))

(defn- text-steps
  [network state epoch-number counter]
  (let [steps (case network
                :mainnet
                {:pending    (mainnet-label 0)
                 :sending     (mainnet-label  (calculate-counter counter))
                 :confirmed   (mainnet-label  (calculate-counter counter))
                 :finalising (mainnet-label  (calculate-counter counter))
                 :finalized  (i18n/label :t/epoch-number {:number epoch-number})
                 :error      (mainnet-label 0)}
                (or :optimism :arbitrum)
                {:pending    (subnet-label 0)
                 :sending   (subnet-label 0)
                 :confirmed  (subnet-label 0)
                 :finalising (subnet-label 1)
                 :finalized  (i18n/label :t/epoch-number {:number epoch-number})
                 :error      (subnet-label 0)}
                nil)]
    (get-in steps [state])))
```

Reply:
btw @mmilad75 is `counter` passed as a ratom into the component? Do we actually update this ratom inside this component? 🤔 if not then we should definitely be passing in the deref'd value to the component and this component should not need to know whether it is a ratom or not. It should just take a value for counter 👍

Reply:
Done